spec/domain/repository/persist_repository_from_upstream_spec.rb

```diff
@@ -47,41 +47,74 @@
 
       expect(Repository.find(existing_repository.id).attributes.symbolize_keys).to include(**expected_attributes)
     end
+
+    context "with name changed in upstream data" do
+      let(:full_name) { "test/name-change" }
+      let(:raw_upstream_data) { build(:raw_upstream_data, full_name: full_name) }
+      let(:existing_repository) { create(:repository, full_name: full_name) }
+      let(:incoming_repository) { build(:repository) }
+
+      context "upstream update fails" do
+        before do
+          allow(RepositoryHost::Github).to receive(:fetch_repo).with(existing_repository.id_or_name, nil).and_return(nil)
+        end
+
+        it "deletes clashed name" do
+          described_class.update_from_host_data(incoming_repository, raw_upstream_data)
+          expect(Repository.find_by(id: existing_repository.id)).to be_nil
```

Comment:
🎱 Could be worth putting in a changed by -1 matcher, though that's testing the setup a bit (making sure the setup also created the object & got deleted, as opposed to the object never existing in the first place)